src/plus/focus/focusProvider.ts

```diff
@@ -313,230 +328,86 @@ export class FocusProvider implements Disposable {
-										categorized.push(this.createFocusItem('waiting-for-review', pr, enrichedItem));
-										continue outer;
-									} else {
-										categorized.push(this.createFocusItem('waiting-for-review', pr, enrichedItem));
-										continue outer;
-									}
-									break;
-							}
-							break;
-						case PullRequestMergeableState.Conflicting:
-							if (
-								pr.pullRequest.reviewDecision === PullRequestReviewDecision.Approved &&
-								viewerHasMergeAccess
-							) {
-								categorized.push(this.createFocusItem('mergeable-conflicts', pr, enrichedItem));
-							} else {
-								categorized.push(this.createFocusItem('conflicts', pr, enrichedItem));
-							}
-							continue outer;
-					}
-				}
-
-				if (pr.reasons.includes('review-requested')) {
-					// Skip adding if there are failed CI checks
-					if (pr.pullRequest.statusCheckRollupState === PullRequestStatusCheckRollupState.Failed) continue;
-
-					categorized.push(this.createFocusItem('needs-review', pr, enrichedItem));
-					continue;
+		const enrichedItems = getSettledValue(enrichedItemsResult);
+		// Multiple enriched items can have the same entityId. Map by entityId to an array of enriched items.
+		const enrichedItemsByEntityId: { [id: string]: EnrichedItem[] } = {};
+		if (enrichedItems != null) {
+			for (const enrichedItem of enrichedItems) {
+				if (enrichedItem.entityId in enrichedItemsByEntityId) {
+					enrichedItemsByEntityId[enrichedItem.entityId].push(enrichedItem);
+				} else {
+					enrichedItemsByEntityId[enrichedItem.entityId] = [enrichedItem];
 				}
 			}
 		}
 
-		// const issues = getSettledValue(issuesResult);
-		// if (issues != null) {
-		// 	for (const issue of issues.splice(0, 3)) {
-		// 		let next = false;
-
-		// 		const enrichedItem = enrichedItems.get(issue.issue.nodeId!);
-		// 		if (enrichedItem != null) {
-		// 			switch (enrichedItem.type) {
-		// 				case 'pin':
-		// 					addItemToGroup(grouped, 'Pinned', issue);
-		// 					next = true;
-		// 					break;
-		// 				case 'snooze':
-		// 					addItemToGroup(grouped, 'Snoozed', issue);
-		// 					next = true;
-		// 					break;
-		// 			}
-
-		// 			if (next) continue;
-		// 		}
-
-		// 		if (issue.reasons.includes('assigned')) {
-		// 			addItemToGroup(grouped, 'In Progress', issue);
-		// 			continue;
-		// 		}
-		// 	}
-		// }
-
-		// // Sort the grouped map by the order of the Groups array
-		// const sorted = new Map<FocusActionCategory, FocusItem[]>();
-		// for (const group of actionCategories) {
-		// 	const items = categorized.get(group);
-		// 	if (items == null) continue;
-
-		// 	sorted.set(
-		// 		group,
-		// 		items.sort((a, b) => (a.pinned ? -1 : 1) - (b.pinned ? -1 : 1) || b.sortTime - a.sortTime),
-		// 	);
-		// }
-
-		this.updateGroupedIds(categorized);
-		this._onDidRefresh.fire({ items: categorized });
-		return categorized;
-	}
+		if (prs != null) {
+			const github = await this.container.integrations.get(HostingIntegrationId.GitHub);
```

Comment:
Can we move this up to be in parallel with the PR request (its too bad we need to make another call, since we have the viewer information when we make the PR request)

Reply:
We have to use an `as` somewhere in here because even though the shared provider library function is expecting a certain shape for PRs, we are adding additional fields to the inputs which we expect to see on the output. To prevent TS from thinking we lost those fields, we need to cast until the util function typing can be updated.

For now I will use `satisfies` here and move the `as` up to the output of the `getActionablePullRequests` call to be more accurate with where the issue is.

Reply:
I have it here to avoid making the extra api call unless we actually have PRs that we want to categorize. If I move it up with the PR request we always have an extra api call even when the output isn't used.